### PR TITLE
Feature/i18n Support & Content Customization

### PR DESCRIPTION
**Pull request by emclaug2**
#### Changes proposed in this Pull Request:
- Split docs up into smaller sections
- Adds /translations folder with english & french source files. 
- Make it so Error empty states can be customized with custom content. 
- Add `languageCode` prop to PxbAuthConfig which determines which language source to use. 

These changes don't use ngx-translate or any i18n switching dependency. 

Users can edit these translation files by doing the following:

```
import { pxbAuthEnglishTranslations, PxbAuthTranslations, mergeAuthTranslations }  from '@pxblue/angular-auth-workflow';

ngOnInit(): void {
    // For single text element changes: 
    pxbAuthEnglishTranslations.GENERAL.BACK_BUTTON = 'Go Back';
    
    // For multi-line or multi-language text overrides:
    const authEnglishOverrides: PxbAuthTranslationsOverride = {
        GENERAL: { ... }
        LOGIN: { ... },
        CREATE_ACCOUNT: { ... },
    };
    mergeAuthTranslations(pxbAuthEnglishTranslations, authEnglishOverrides);
}
```

### CODE REVIEW

**Comment by emclaug2** [comment]
I'm still figuring out the best way to handle password requirement customization.  This doesn't not work with i18n yet.

**Comment by emclaug2** [comment]
> I'm still figuring out the best way to handle password requirement customization. This doesn't work with i18n yet.

I had to change the way we handle password requirement customizations, luckily this was already a breaking change. 

This is new in the `PxbAuthConfig`:

```
    /* Password requirements */
    defaultPasswordRequirements = {
        characterLimit: true,
        uppercaseLetter: true,
        lowercaseLetter: true,
        requireNumber: true,
        specialCharacter: true,
    };
    customPasswordRequirements: PasswordRequirement[] = [];
```

Password requirements are now generated on page-render instead of on app-loads.  This allows us to dynamically change requirements' text on a language change.

**Comment by emclaug2** [comment]
Hosted here: https://pxblue-temp.web.app/auth/login

I added a language selector in debug mode to see Login page translations in FR.  I didn't add all french translations.

**Comment by emclaug2** [comment]
I noticed that I was unable to login to the home page because the ON_AUTHENTICATED route was not being set correctly.  

This PR now fixes #52 

PxbAuthSecurityService has a new method `inferOnAuthenticatedRoute` (optional) to be called on app-load to set ON_AUTHENTICATED route by parsing the URL.   This'll make sure we end up in the correct place after a user logs in.

**Comment by daileytj** [comment]
There's an issue with the translation on the forgot password confirmation page. Here are some screenshots, one from running it locally and one from the app you put up on firebase.

Screen Shot 2021-02-18 at 1 56 24 PM
Screen Shot 2021-02-18 at 1 59 30 PM

**Comment by emclaug2** [comment]
Good catch @daileytj, it should be fixed now.

I went down a rabbit hole this afternoon and did some refactoring on how we manage email & password form fields.  I made a component wrapper to make things more consistent between pages.

File: login-workflow/docs/custom-account-details.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
+# Custom Account Details
+The `@pxblue/angular-auth-workflow` allows for custom account detail fields to be added during a user's self-registration.   To add custom form fields to the self registration pages, follow the steps below:
```
Comment (joebochill) [inline login-workflow/docs/custom-account-details.md:2]: You got a little overzealous with the spacebar here lol
```suggestion
The `@pxblue/angular-auth-workflow` allows for custom account detail fields to be added during a user's self-registration. To add custom form fields to the self registration pages, follow the steps below:
```

File: login-workflow/docs/custom-account-details.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+```
+
+### Populate the `accountDetails` @Input
+
+Each `accountDetails` object has 3 properties:
+
+| Property            | Description                                            |
+| ------------------- | ------------------------------------------------------ | 
+| form                | What to render                                         |
+| formControls        | the login screen                                       |
```
Comment (joebochill) [inline login-workflow/docs/custom-account-details.md:36]: Is this right?

File: login-workflow/docs/custom-account-details.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+</ng-template>
+```
+
+### Populate the `accountDetails` @Input
+
+Each `accountDetails` object has 3 properties:
+
+| Property            | Description                                            |
+| ------------------- | ------------------------------------------------------ | 
+| form                | What to render                                         |
```
Comment (joebochill) [inline login-workflow/docs/custom-account-details.md:35]:
```suggestion
| form                | template ref of the form                                         |
```

File: login-workflow/docs/custom-account-details.md (hunk 2 above)
Comment (joebochill) [reply login-workflow/docs/custom-account-details.md:36]: should it say something like 'a map listing all of the fields in the form'

File: login-workflow/docs/page-customization.md [hunk 4]
```diff
@@ -0,0 +1,175 @@
+# Page Customizations
+ 
+Each page within the `@pxblue/angular-auth-workflow` can be customized with string `@Inputs` or `ng-content`.  
+ 
+## Text and HTML Overrides
+
+Each text element within the auth workflow has the ability to be overwritten.
```
Comment (joebochill) [inline login-workflow/docs/page-customization.md:7]: Add a link to the file with the available string definitions ?

**Review by joebochill — APPROVED** [review]

File: login-workflow/docs/custom-account-details.md (hunk 2 above)
Comment (emclaug2) [reply login-workflow/docs/custom-account-details.md:36]: Good catch, this must have been copied/pasted over incorrectly.

**Pull request merged**